samples/bot-sso-adaptivecard/csharp/BotSsoAdaptivecard/Bots/DialogBot.cs

```diff
@@ -0,0 +1,275 @@
+                if (value["state"] != null)
+                {
+                    state = value["state"].ToString();
+                }
+
+                // authToken and state are absent, handle verb
+                if (authentication == null && state == null)
+                {
+                    switch (verb)
+                    {   //when token is absent in the invoke. We can initiate SSO in response to the invoke
+                        case "initiateSSO":
+                            return await initiateSSOAsync(turnContext, cancellationToken);
+                    }
+                }
+                else
+                {
+                    return createAdaptiveCardInvokeResponseAsync(authentication, state);
+
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Authentication success.
+        /// AuthToken or state is present. Verify token/state in invoke payload and return AC response
+        /// </summary>
+        /// <param name="authentication"></param>
+        /// <param name="state"></param>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <param name="isBasicRefresh"></param>
+        /// <param name="fileName"></param>
+        /// <returns></returns>
+        private InvokeResponse createAdaptiveCardInvokeResponseAsync(JObject authentication, string state, bool isBasicRefresh = false, string fileName = "adaptiveCardResponseJson.json")
+        {
+            //verify token is present or not
+
+            bool isTokenPresent = authentication != null ? true : false;
+            bool isStatePresent = state != null && state != "" ? true : false;
+
+            // TODO : Use token or state to perform operation on behalf of user
+
+            string[] filepath = { ".", "Resources", fileName };
+
+            var adaptiveCardJson = File.ReadAllText(Path.Combine(filepath));
+            AdaptiveCardTemplate template = new AdaptiveCardTemplate(adaptiveCardJson);
+            var authResultData = isTokenPresent ? "SSO success" : isStatePresent ? "OAuth success" : "SSO/OAuth failed";
+            if (isBasicRefresh)
+            {
+                authResultData = "Refresh done";
+            }
+            var payloadData = new
+            {
+                authResult = authResultData,
+            };
+
+            var cardJsonstring = template.Expand(payloadData);
+
+            var adaptiveCardResponse = new AdaptiveCardInvokeResponse()
+            {
+                StatusCode = 200,
+                Type = AdaptiveCard.ContentType,
+                Value = JsonConvert.DeserializeObject(cardJsonstring)
+            };
+            return CreateInvokeResponse(adaptiveCardResponse);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
```

Comment:
No explanation added for these parameters

Reply:
Fixed